src/app/app.component.ts

```diff
@@ -70,6 +83,27 @@ export class AppComponent implements OnInit, OnDestroy {
     }
 
     ngOnInit() {
+        this.statusSubject
+            .pipe(
+                switchMap(() => this.raidenService.getStatus()),
+                tap((status) => {
+                    this.apiStatus = status;
+                    if (status.status === 'syncing') {
+                        if (this.initialBlocksToSync === undefined) {
+                            this.initialBlocksToSync = status.blocks_to_sync;
+                        }
+                        this.syncingProgress =
+                            (this.initialBlocksToSync - status.blocks_to_sync) /
+                            this.initialBlocksToSync;
+                    }
+                }),
+                takeUntil(this.ngUnsubscribe),
+                takeWhile((status) => status.status !== 'ready'),
+                delay(500),
+                tap(() => this.statusSubject.next(null))
```

Comment:
Nitpick (i.e. just a headsup, no need to change here/now): `tap`ing last would be more descriptive if it'd be put in `subscribe` cb instead

Reply:
Or here, as it's `next`ing to a `Subject`, could also be something like `..., mapTo(null)).subscribe(subject)`; but mind that this would also forward error/completes to the subject

Reply:
I never used a Subject as an Observer, but now I see when it is useful. Thanks a lot!